File: 3637. Trionic Array I.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) 
    {
        int i = 0, n = nums.size();
        bool flag1 = false, flag2 = false, flag3 = false;
        while(i < n - 1 and nums[i] < nums[i + 1]) flag1 = true; i++;            
        while(i < n - 1 and nums[i] > nums[i + 1]) flag2 = true; i++;
        while(i < n - 1 and nums[i] < nums[i + 1]) flag3 = true; i++;
        return i == n - 1 and flag1 and flag2 and flag3;
    }
};

File: 3637. Trionic Array I.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        i, n = 0, len(nums)
        f1, f2, f3 = 0, 0, 0
        
        while i < n - 1 and nums[i] < nums[i + 1]:
            f1 = 1
            i += 1
        
        while i < n - 1 and nums[i] > nums[i + 1]:
            f2 = 1
            i += 1
        
        while i < n - 1 and nums[i] < nums[i + 1]:
            f3 = 1
            i += 1
        
        return bool(i == n - 1 and f1 and f2 and f3)